File: src/ua/edu/hneu/oop/dotNet/LR6/LR6/CarFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LR6
{
    public abstract class CarFactory
    {
        public abstract AbstractCar CreateCar();
        public abstract AbstractEngine CreateEngine();
    }

    class AudiFactory : CarFactory
    {
        public override AbstractCar CreateCar()
        {
            return new AudiCar();
        }

        public override AbstractEngine CreateEngine()
        {
            return new AudiEngine();
        }
    }

    class BMWFactory : CarFactory
    {
        public override AbstractCar CreateCar()
        {
            return new BMWCar();
        }

        public override AbstractEngine CreateEngine()
        {
            return new BMWEngine();
        }
    }
}

File: src/ua/edu/hneu/oop/java/lr6/task1/domain/factory/CarFactory.java
package ua.edu.hneu.oop.java.lr6.task1.domain.factory;

import ua.edu.hneu.oop.java.lr6.task1.domain.entity.AbstractCar;
import ua.edu.hneu.oop.java.lr6.task1.domain.entity.AbstractEngine;

public abstract class CarFactory {
    public abstract AbstractCar createCar();

    public abstract AbstractEngine createEngine();
}
